doc/devel/coding_style_guideline.rst

```diff
@@ -49,6 +49,182 @@ No alias should be used for `h5py`::
+      double *ps = <double *> cnp.PyArray_DATA(seed)
+      cnp.npy_intp *pstr = <cnp.npy_intp *> qa.strides
+      cnp.npy_intp d, i, j, cnt
+      double direction[3]
+      double tmp, ftmp
+  cdef int true_enough(x):
+      return 1
+
+  # Bad
+  cdef  char *s
+  cdef char * s, * t, * u, * v
+  cdef double Xf[3], d[3]
+  cdef double x=42, y=x+1, z=x*y
+  cdef Material     mat   = Material()
+  cdef   int   falsified(x):
+      return 0
+
+Inside of a function, place all `cdef` statements at the top of the function
+body::
+
+  # Good
+  cdef int true_enough(x):
+      cdef int i = x
+      cdef int rtn
+      rtn = i + 42
+      return rtn
+
+  # Bad
+  cdef int falsified(x):
+      cdef int i = x, j = -42
+      j += i
+      cdef int rtn = j / j - 1
+      return rtn
+
+Using C libraries
+-----------------
+The `cimport`s should follow the same rules defined in PEP8 for `import`
+statements. If a module is both *imported* and *cimported*, the `cimport`
+should come before the `import`.
+
+An example of an imported C library::
+
+  from libc.stdlib cimport calloc, realloc, free
+
+Do not use `include` statements.
+
+Error return values
+-------------------
+When declaring an error return value with the `except` keyword, use one space on
+both sides of the `except`. If in a function definition, there should be no
+spaces between the error return value and the colon `:`. Avoid `except *`
+unless it is needed for functions returning `void`::
+
+  # Good
+  cdef void redwood() except *
+  cdef void c_extract(Feature self, Data2D datum, Data2D out) nogil except *:
+  cdef int sequoia(x) except +:
+      ...
+
+  # Bad
+  cdef char * spruce(x) except *:
+  cdef int fir(x)    except   +  :
+      ...
+
+Pointers and references
+-----------------------
+Pointers and references may be either zero or one space away from the type name.
+If followed by a variable name, they must be one space away from the variable
+name. Do not put any spaces between the reference operator `&` and the variable
+name::
+
+  # Good
+  cdef int& i
+  cdef char * s
+  i = &j
+
+  # Bad
+  cdef int &i
+  cdef char *s
+  i = & j
+
+Casting
+-------
+When casting a variable there must be no whitespace between the opening `<` and
+the type. There must one space between the closing `>` and the variable::
+
+  # Good
+  <float> i
+  <void *> s
+
+  # Bad
+  < float >i
+  <void*>  s
+
+Loops
+-----
+Use Python loop syntax::
+
+  for i in range(nrows):
+    ...
+
+Other for-loop constructs are deprecated and must be avoided.
+
+The `property` keyword
+----------------------
+There should be exactly one space between the `property` keyword and the
+attribute name. There may be no spaces between the attribute name and the colon
+`:`. All properties should have docstrings. There should be no blank lines
+between the property declaration line and the following line::
+
+  # Good
+  property is_order_invariant:
+      """ Is this feature invariant to the sequence's ordering """
```

Comment:
Not sure if PEP8 says anything about whitespaces in docstrings.

Reply:
Space after keywords is a pep8 rule. So you can remove this section.

Reply:
Thanks for the review. Removed in ff65019.